receiver/mongodbatlasreceiver/events.go

```diff
@@ -0,0 +1,335 @@
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/extension/experimental/storage"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	rcvr "go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/adapter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal"
+)
+
+const (
+	eventStorageKey       = "last_recorded_event"
+	defaultEventsMaxPages = 25
+	defaultEventsPageSize = 100
+	defaultPollInterval   = time.Minute
+)
+
+type eventsClient interface {
+	GetProject(ctx context.Context, groupID string) (*mongodbatlas.Project, error)
+	GetEvents(ctx context.Context, groupID string, opts *internal.GetEventsOptions) (ret []*mongodbatlas.Event, nextPage bool, err error)
+}
+
+type eventsReceiver struct {
+	client        eventsClient
+	logger        *zap.Logger
+	id            component.ID  // ID of the receiver component
+	storageID     *component.ID // ID of the storage extension component
+	storageClient storage.Client
+	cfg           *Config
+	consumer      consumer.Logs
+
+	maxPages     int
+	pageSize     int
+	pollInterval time.Duration
+	wg           *sync.WaitGroup
+	record       *eventRecord
+	doneChan     chan bool
+}
+
+type eventRecord struct {
+	sync.Mutex
+	LastRecordedEvent *time.Time `mapstructure:"last_recorded_event"`
+}
+
+func newEventsReceiver(settings rcvr.CreateSettings, c *Config, consumer consumer.Logs) *eventsReceiver {
+	r := &eventsReceiver{
+		client:       internal.NewMongoDBAtlasClient(c.PublicKey, c.PrivateKey, c.RetrySettings, settings.Logger),
+		cfg:          c,
+		logger:       settings.Logger,
+		id:           settings.ID,
+		storageID:    c.StorageID,
+		consumer:     consumer,
+		pollInterval: c.Events.PollInterval,
+		wg:           &sync.WaitGroup{},
+		maxPages:     int(c.Events.MaxPages),
+		pageSize:     int(c.Events.PageSize),
+		doneChan:     make(chan bool, 1),
+	}
+
+	if r.maxPages == 0 {
+		r.maxPages = defaultEventsMaxPages
+	}
+
+	if r.pageSize == 0 {
+		r.pageSize = defaultEventsPageSize
+	}
+
+	if r.pollInterval == 0 {
+		r.pollInterval = time.Minute
+	}
+
+	return r
+}
+
+func (er *eventsReceiver) Start(ctx context.Context, host component.Host) error {
+	er.logger.Debug("Starting up events receiver")
+	return er.startPolling(ctx, host)
+}
+
+func (er *eventsReceiver) Shutdown(ctx context.Context) error {
+	er.logger.Debug("Shutting down events receiver")
+	close(er.doneChan)
+	er.wg.Wait()
+	return er.checkpoint(ctx)
+}
+
+func (er *eventsReceiver) startPolling(ctx context.Context, host component.Host) error {
+	storageClient, err := adapter.GetStorageClient(ctx, host, er.storageID, er.id)
+	if err != nil {
+		return fmt.Errorf("failed to set up storage: %w", err)
+	}
+	er.storageClient = storageClient
+	err = er.loadCheckpoint(ctx)
```

Comment:
This isn't really polling related. Let's pull it out into the `Start` method, prior to calling `startPolling`.